catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/AbstractCswSource.java

```diff
@@ -1644,6 +1685,232 @@ public void setMetacardTypes(List<MetacardType> types) {
         this.metacardTypes = types;
     }
 
+    @Override
+    public CreateResponse create(CreateRequest createRequest) throws IngestException {
+        Map<String, Serializable> properties = new HashMap<>();
+
+        Subject subject =
+                (Subject) createRequest.getPropertyValue(SecurityConstants.SECURITY_SUBJECT);
+        Csw csw = factory.getClientForSubject(subject);
+        CswTransactionRequest transactionRequest = new CswTransactionRequest();
+        transactionRequest.setVersion(CswConstants.VERSION_2_0_2);
+        transactionRequest.setService(CswConstants.CSW);
+        transactionRequest.setVerbose(true);
+
+        List<Metacard> metacards = createRequest.getMetacards();
+        ArrayList<Metacard> createdMetacards = new ArrayList<>();
+        ArrayList<Filter> createdMetacardFilters = new ArrayList<>();
+
+        String insertTypeName = schemaTransformerManager.getTransformerIdForSchema(
+                cswSourceConfiguration.getOutputSchema());
+
+        if (insertTypeName == null) {
+            insertTypeName = CswConstants.CSW_RECORD;
+        }
+
+        transactionRequest.getInsertActions()
+                .add(new InsertAction(insertTypeName, null, metacards));
+        try {
+            TransactionResponseType response = csw.transaction(transactionRequest);
+            if (response.getTransactionSummary()
+                    .getTotalInserted()
+                    .longValue() != createRequest.getMetacards()
+                    .size()) {
+                throw new CswException("One or more inserts failed");
+            }
+
+            //dive down into the response to get the created ID's. We need these so we can query
+            //the source again to get the created metacards and put them in the result
+            for (InsertResultType record : response.getInsertResult()) {
+                for (BriefRecordType brief : record.getBriefRecord()) {
+                    for (JAXBElement<SimpleLiteral> identifier : brief.getIdentifier()) {
+                        for (String id : identifier.getValue()
+                                .getContent()) {
+                            createdMetacardFilters.add(filterBuilder.attribute(Metacard.ID)
+                                    .is()
+                                    .equalTo()
+                                    .text(id));
+                        }
+                    }
+                }
+            }
+
+        } catch (CswException e) {
+            LOGGER.debug("Csw Transaction Failed. ", e);
+            throw new IngestException("Csw Transaction Failed : " + e.getMessage());
```

Comment:
Throwing here doesn't make sense. At this point some records were successfully ingested in the CatalogStore. You should add the failures to the processing details including the IDs so the caller knows what failed.